examples/query-complexity/recipe-resolver.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  Resolver,
+  Query,
+  FieldResolver,
+  Arg,
+  Root,
+  Mutation,
+  Float,
+  Int,
+  ResolverInterface,
+} from "../../src";
+import { plainToClass } from "class-transformer";
+
+import { Recipe } from "./recipe-type";
+import { RecipeInput } from "./recipe-input";
+import { createRecipeSamples } from "./recipe-samples";
+
+@Resolver(of => Recipe)
+export class RecipeResolver implements ResolverInterface<Recipe> {
+  private readonly items: Recipe[] = createRecipeSamples();
+
+  @Query(returns => Recipe, { nullable: true })
+  async recipe(@Arg("title") title: string): Promise<Recipe | undefined> {
+    return await this.items.find(recipe => recipe.title === title);
+  }
+
+  @Query(returns => [Recipe], { description: "Get all the recipes from around the world " })
+  async recipes(): Promise<Recipe[]> {
+    return await this.items;
+  }
+
+  @Mutation(returns => Recipe, { complexity: 10 })
```

Comment:
You should remove all the bloat from this example - it should focus only on complexity, so single query returned object with nested fields, array of getter is enought, no need to polute it with mutation and inputs 😉